sherpa/astro/data.py

```diff
@@ -163,6 +163,9 @@ def _validate_energy_ranges(self, label, elo, ehi, ethresh):
 
         rtype = self._ui_name
 
+        if elo.size != ehi.size:
+            raise ValueError("rmflo and rmfhi must have the same size, not {} and {}".format(elo.size, ehi.size))
```

Comment:
I suggest changing this message to

```
"The energy arrays must have the same size, not {} and {}" .format(elo.size, ehi.size)
```

since the "useful" name of the arrays (that is, the input column name or function argument name) is "long lost" at this point.